extensions/json-language-features/client/src/jsonClient.ts

```diff
@@ -163,25 +164,26 @@ export async function startClient(context: ExtensionContext, newLanguageClient:
 		window.showInformationMessage(l10n.t('JSON schema cache cleared.'));
 	}));
 
+	toDispose.push(workspace.onDidSaveTextDocument(async (document) => {
```

Comment:
To be a good save participant you shouldn't be doing work on _did_ save but during _will_ save. That's because otherwise you are triggering a save right after it did save. Also make sure to register the promise with `TextDocumentWillSaveEvent#waitUntil`